datadog/tests/resource_datadog_security_monitoring_filter_test.go

```diff
@@ -0,0 +1,159 @@
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: accProviders,
+		CheckDestroy:      testAccCheckDatadogSecurityMonitoringFilterDestroy(accProvider),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckDatadogSecurityMonitoringFilterCreated(filterName),
+				Check:  testAccCheckDatadogSecurityMonitorFilterCreatedCheck(accProvider, filterName),
+			},
+			{
+				Config: testAccCheckDatadogSecurityMonitoringFilterUpdated(filterName),
+				Check:  testAccCheckDatadogSecurityMonitorFilterUpdatedCheck(accProvider, filterName),
+			},
+		},
+	})
+}
+
+func testAccCheckDatadogSecurityMonitoringFilterCreated(name string) string {
+	return fmt.Sprintf(`
+resource "datadog_security_monitoring_filter" "acceptance_test" {
+    name = "%s"
+    query = "first query"
+    is_enabled = true
+
+    exclusion_filter {
+        name = "first"
+        query = "does not really match much"
+    }
+
+    exclusion_filter {
+        name = "second"
+        query = "neither does it"
+    }
+}
+`, name)
+}
+
+func testAccCheckDatadogSecurityMonitorFilterCreatedCheck(accProvider func() (*schema.Provider, error), filterName string) resource.TestCheckFunc {
+	return resource.ComposeTestCheckFunc(
+		testAccCheckDatadogSecurityMonitoringFilterExists(accProvider),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "name", filterName),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "query", "first query"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "is_enabled", "true"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "exclusion_filter.0.name", "first"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "exclusion_filter.0.query", "does not really match much"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "exclusion_filter.1.name", "second"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "exclusion_filter.1.query", "neither does it"),
+	)
+}
+
+func testAccCheckDatadogSecurityMonitoringFilterUpdated(name string) string {
+	return fmt.Sprintf(`
+resource "datadog_security_monitoring_filter" "acceptance_test" {
+    name = "%s"
+    query = "new query"
+    is_enabled = false
+
+    exclusion_filter {
+        name = "first"
+        query = "does not really match much"
+    }
+
+    exclusion_filter {
+        name = "third"
+        query = "I am new"
+    }
+}
+`, name)
+}
+
+func testAccCheckDatadogSecurityMonitorFilterUpdatedCheck(accProvider func() (*schema.Provider, error), filterName string) resource.TestCheckFunc {
+	return resource.ComposeTestCheckFunc(
+		testAccCheckDatadogSecurityMonitoringFilterExists(accProvider),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "name", filterName),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "query", "new query"),
+		resource.TestCheckResourceAttr(
+			tfSecurityFilterName, "is_enabled", "false"),
+		resource.TestCheckResourceAttr(
```

Comment:
You should add a check for the length, in `exclusion_filter.#`.

Reply:
Thanks for all the comment; the code looks much cleaner now. I'll update the rule ressource too with this in mind.